model/swagger.yml

```diff
@@ -37,14 +61,218 @@ paths:
+      - $ref: "#/components/parameters/Sort"
+      - $ref: "#/components/parameters/Start"
+      - $ref: "#/components/parameters/Versions"
+  /classes/{class}/{identifier}/member-of:
+    get:
+      tags:
+        - references
+      summary: |
+        returns one or more PDS Products that have the given PDS product class and lid/lidvid as a member.
+      operationId: class-member-of
+      responses:
+        '200':
+          $ref: "#/components/responses/Plural"
+        '400':
+          $ref: "#/components/responses/Error"
+        '404':
+          $ref: "#/components/responses/Error"
+        '500':
+          $ref: "#/components/responses/Error"
+        '501':
+          $ref: "#/components/responses/Error"
+    parameters:
+      - $ref: "#/components/parameters/Fields"
+      - $ref: "#/components/parameters/Class"
+      - $ref: "#/components/parameters/Identifier"
+      - $ref: "#/components/parameters/Limit"
+      - $ref: "#/components/parameters/Sort"
+      - $ref: "#/components/parameters/Start"
+  /classes/{class}/{identifier}/member-of/{versions}:
+    get:
+      tags:
+        - references
+      summary: |
+        returns one or more PDS Products that have the given PDS product class and lid/lidvid as a member.
+      operationId: class-member-of-vers
+      responses:
+        '200':
+          $ref: "#/components/responses/Plural"
+        '400':
+          $ref: "#/components/responses/Error"
+        '404':
+          $ref: "#/components/responses/Error"
+        '500':
+          $ref: "#/components/responses/Error"
+        '501':
+          $ref: "#/components/responses/Error"
+    parameters:
+      - $ref: "#/components/parameters/Fields"
+      - $ref: "#/components/parameters/Class"
+      - $ref: "#/components/parameters/Identifier"
+      - $ref: "#/components/parameters/Limit"
+      - $ref: "#/components/parameters/Sort"
+      - $ref: "#/components/parameters/Start"
+      - $ref: "#/components/parameters/Versions"
+  /classes/{class}/{identifier}/member-of/member-of:
+    get:
+      tags:
+        - references
+      summary: |
+        returns one or more PDS Products that have the given PDS product class and lid/lidvid as a member of its members.
+      operationId: class-member-of-of
+      responses:
+        '200':
+          $ref: "#/components/responses/Plural"
+        '400':
+          $ref: "#/components/responses/Error"
+        '404':
+          $ref: "#/components/responses/Error"
+        '500':
+          $ref: "#/components/responses/Error"
+        '501':
+          $ref: "#/components/responses/Error"
+    parameters:
+      - $ref: "#/components/parameters/Fields"
+      - $ref: "#/components/parameters/Class"
+      - $ref: "#/components/parameters/Identifier"
+      - $ref: "#/components/parameters/Limit"
+      - $ref: "#/components/parameters/Sort"
+      - $ref: "#/components/parameters/Start"
+  /classes/{class}/{identifier}/member-of/member-of/{versions}:
```

Comment:
Is there somewhere in the API docs we could define the value-set for `{versions}`?

Reply:
@al-niessner ☝️

Reply:
@jordanpadams

values for {versions} is defined by the yaml class Versions. To find it easily, look for Versions: and you will see that it maps to {versions} via its name and the enum choices. The docs should show it correctly as all or latest.

Because you are going to want it, it cannot be blank. It is why you have to have two nearly identical endpoints of /products/{identifier}/members and /products/{identifier}/members/{versions}.

/products/{identifier} cannot use {versions} because the all/latest changes the output between Plural/Singular respectively - plural and singular are known a priori. In the members and members-of cases they are always Plural - count can only be known a posteriori - so {versions} works.